snuba/datasets/processors/replays_processor.py

```diff
@@ -185,25 +185,76 @@ def process_message(
-            self._process_user(processed, replay_event)
-            self._process_event_hash(processed, replay_event)
-            self._process_contexts(processed, replay_event)
-            return InsertBatch([processed], None)
+            if replay_event["type"] == "replay_actions":
+                actions = process_replay_actions(replay_event, processed, metadata)
+                return InsertBatch(actions, None)
+            else:
+                # The following helper functions should be able to be applied in any order.
+                # At time of writing, there are no reads of the values in the `processed`
+                # dictionary to inform values in other functions.
+                # Ideally we keep continue that rule
+                self._process_base_replay_event_values(processed, replay_event)
+                self._process_tags(processed, replay_event)
+                self._process_sdk(processed, replay_event)
+                self._process_kafka_metadata(metadata, processed)
+
+                # the following operation modifies the event_dict and is therefore *not*
+                # order-independent
+                self._process_user(processed, replay_event)
+                self._process_event_hash(processed, replay_event)
+                self._process_contexts(processed, replay_event)
+                return InsertBatch([processed], None)
         except Exception:
             metrics.increment("consumer_error")
             raise
 
 
+def process_replay_actions(
+    payload: Mapping[Any, Any],
+    processed: Mapping[Any, Any],
+    metadata: KafkaMessageMetadata,
+) -> list[dict[str, Any]]:
+    """Process replay_actions message type."""
+    return [
+        {
+            # Primary-key.
+            "project_id": processed["project_id"],
+            "timestamp": default(
+                lambda: datetime.now(timezone.utc),
+                maybe(to_datetime, payload.get("timestamp")),
+            ),
+            "replay_id": to_uuid(payload["replay_id"]),
+            "segment_id": None,
+            "event_hash": payload.get("event_hash", segment_id_to_event_hash(None)),
+            # Default values for non-nullable columns.
+            "trace_ids": [],
+            "error_ids": [],
+            "urls": [],
+            "title": None,
+            "platform": "javascript",
+            "user": None,
+            "sdk_name": None,
+            "sdk_version": None,
+            # Kafka columns.
+            "retention_days": processed["retention_days"],
+            "partition": metadata.partition,
+            "offset": metadata.offset,
+            # DOM Index fields.
+            "click_node_id": _collapse_or_err(_collapse_uint32, int(click["node_id"])),
+            "click_tag": to_string(click["tag"])[:32],
+            "click_id": to_string(click["id"])[:64],
+            "click_class": to_typed_list(to_string, click["class"][:10]),
```

Comment:
This is limiting just the number of elements right? Since this is arbitrary user input I would cap each element as well.